.github/scripts/roll_sdk_packages.sh

```
@@ -0,0 +1,354 @@
+      log_info "Running flutter pub upgrade --major-versions in $PROJECT_NAME (all packages)"
+      if ! flutter pub upgrade --major-versions; then
+        log_error "Failed to upgrade all packages in $PROJECT_NAME"
+        cd "$REPO_ROOT"
+        continue
+      fi
+    else
+      log_info "Running flutter pub upgrade for SDK packages only in $PROJECT_NAME"
+      # Upgrade all SDK packages at once
+      if [ ${#SDK_PACKAGES_FOUND[@]} -gt 0 ]; then
+        log_info "Upgrading packages: ${SDK_PACKAGES_FOUND[*]}"
+        if ! flutter pub upgrade --unlock-transitive ${SDK_PACKAGES_FOUND[@]}; then
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)